Add Release Team selection criteria

justaugustus opened the pull request:
Let's lay out some of the details about selecting a Release Team...

This covers:
- selecting leads
- selecting the Release Team Lead
- building a pool of role shadows
- timing
- qualities of a Release Team member
- promoting diversity

/assign @tpepper @AishSundar @jdumars @calebamiles @jberkus 
/sig release
/hold

rel: https://github.com/kubernetes/sig-release/pull/226
Signed-off-by: Stephen Augustus <foo@agst.us>

nikhita commented on release-team/README.md:
```diff
@@ -188,6 +188,60 @@ For all unplanned or embargoed releases
+- training and selecting a successor from non-Release Team members
+- staffing the role themselves
+
+Ultimately, if not of these can be satisfied, responsibility falls to the future Release Team Lead and SIG Release to staff the roles.
```
s/not/none

nikhita commented on release-team/README.md:
```diff
@@ -188,6 +188,60 @@ For all unplanned or embargoed releases
 
 ---
 
+## Release Team Selection
```
nit: can we get a ToC at the top of this document? :)

justaugustus replied:
@nikhita -- agreed that it's a long doc! I've added a short ToC for the high-level things and there's also https://github.com/kubernetes/sig-release/issues/238 opened for cleaning up the role definitions. :)

justaugustus commented:
...because I've never used the command before:
/joke

guineveresaenger commented:
This LGTM but I'll let @nikhita do the honors. <3

nikhita commented on release-team/README.md:
```diff
@@ -188,6 +195,60 @@ For all unplanned or embargoed releases
+
+### Shadows
+
+Every Release Team role should seek to accomodate a set of role shadows. This creates a new avenue for code and non-code contributors alike to contribute to the project. Additionally, it seeds future release cycles with new leaders.
```
s/accomodate/accommodate?

nikhita commented on release-team/README.md:
```diff
@@ -188,6 +195,60 @@ For all unplanned or embargoed releases
+- ethnicity
+- nationality
+- locality
+- company affliation
```
s/affliation/affiliation

nikhita commented:
> This LGTM but I'll let @nikhita do the honors. <3

:heart: 

Looks good but I'll let @tpepper sign off. :)

nikhita commented:
/joke

jberkus commented:
There's material on selecting a Team in the Release Lead guide.  Mind incorporating that material into this doc, and maybe removing the redundant parts from the other guide?

guineveresaenger commented:
/meow

jberkus commented:
/woof

jberkus commented:
Oh.

I just realized that my updates to the release team guide from 8 weeks ago were never merged, which is why you're not seeing the half-page set of instructions on choosing a release team there.  :-(

justaugustus commented:
@jberkus -- I ensured most of what you wrote on https://github.com/kubernetes/sig-release/pull/226 was reflected here. I'll combined the two PRs and incorporate the feedback. :)

AishSundar commented on release-team/README.md:
```diff
@@ -188,6 +195,60 @@ For all unplanned or embargoed releases
+- Try to ensure potential Release Team candidates:
+  - can commit to the amount of time required across the release cycle
+  - are enthusiastic about being on the release team
+  - are members of the relevant SIGs for their role, if applicable
```
Have we been following this requirement in prior releases?

jberkus replied:
I did for 1.11.

AishSundar commented:
/lgtm

idvoretskyi commented on release-team/README.md:
```diff
@@ -188,6 +195,60 @@ For all unplanned or embargoed releases
+- the current pool of Release Team members
+- former Release Team members
+
+_The new Release Team Lead should be agreed by quorum of the current Release Team members._
```
More precise definition of the quorum is likely here.

jberkus replied:
Suggestion:

The new Release Team Lead may be nominated by any SIG-Release member, and will be selected by vote at a regular SIG-Release meeting before the end of the prior release cycle.  In the event that SIG-Release is not holding regular meetings, the prior Release Team Lead may convene an emergency SIG-Release meeting to select the new Release Team Lead by simple majority vote.

idvoretskyi replied:
> The new Release Team Lead may be nominated by any SIG-Release member

I'm not sure if we have a clear definition of the "SIG-Release member".

My suggestions would be "by any Kubernetes community representative, who is a member of the [Kubernetes organization on GitHub](https://github.com/orgs/kubernetes/people)". It solves two questions here:
- Kubernetes release process is the project-wide, not SIG-wide activity.
- We have a formal list of Kubernetes org members on GitHub, we don't have one for SIG-Release (what is fine, because SIG-related activities do not set any formal barrier as org membership does).

We have to work on the voting/election process more precisely, but it's not a blocker for the current PR.

jberkus replied:
Yes: vote to merge this, then hammer out selection/endorsement process at the next SIG-release meeting.

justaugustus replied:
I'll add my $0.02 in, just so it's on record...

The Release Team is a subproject of SIG Release. They should have the agency to make decisions regarding the subproject. Any tiebreaking / contentious resolutions would fallback to the SIG Release Chairs, but only if required.

That said, the selection process should happen within the Release Team by the current Release Team members. 

This solves the question of "What constitutes SIG Release membership?" in the context of the Release Team.

SIG members are free to come and go as they please, so I'm not sure we should strive for a formal definition of a SIG Release member, outside of what is defined according to the contributor ladder.

idvoretskyi commented on release-team/README.md:
```diff
@@ -188,6 +195,60 @@ For all unplanned or embargoed releases
+
+_The new Release Team Lead should be agreed by quorum of the current Release Team members._
+
+**If for some reason a new RT Lead cannot be selected by these methods, the current Release Team Lead should be prepared to continue to serve in the following release cycle until a successor can be found.**
```
What is the deadline to select a new RT TL?

jberkus replied:
Preferred: start of Code Freeze for prior release.

Fall-back: within 3 days after the prior release.

jberkus replied:
However ... I'm concerned that this set of rules makes the Release Lead responsible for a successor, without giving them the power to choose this successor in the event that it's difficult or impossible to convene a quorum of SIG-release members.  This argues either for a really loose definition of "quorum", or giving the RL an out where they can select a successor by fiat if there is no quorum.

idvoretskyi commented on release-team/README.md:
```diff
@@ -188,6 +195,60 @@ For all unplanned or embargoed releases
+  - are members of the relevant SIGs for their role, if applicable
+  - have some prior experience with contributing to Kubernetes (such as shadowing a role for a prior release)
+
+Most importantly, strive for diversity in:
```
+1!

jberkus commented:
per discussion on related PR,

/lgtm

justaugustus commented:
@tpepper @AishSundar @jberkus @idvoretskyi -- I've incorporated all of the lovely feedback across this PR, #226, and #261, so I think we're ready for a final round of reviews.

...and perhaps one more kitteh.
/meow

calebamiles commented on release-team/README.md:
```diff
@@ -188,6 +195,69 @@ For all unplanned or embargoed releases
+
+Bear in mind that these are suggestions based on precedent and a Release Team Lead may be nominated by any Release Team member, past or present.
+
+_The new Release Team Lead should be agreed by a majority of the current Release Team members._
```
I think lazy consensus among the previous release team members has worked perfectly well here, but I might note that in the failure to select a release team lead will escalate the decision and/or staffing requirement to the SIG chairs.

calebamiles commented on release-team/README.md:
```diff
@@ -188,6 +195,69 @@ For all unplanned or embargoed releases
+
+_The new Release Team Lead should be agreed by a majority of the current Release Team members._
+
+**If for some reason a new RT Lead cannot be selected by these methods, the current Release Team Lead should be prepared to continue to serve in the following release cycle until a successor can be found.**
```
I would strike this given the above. We did actually try to have this as a requirement but it was deemed unfair to the release team lead who shouldn't really be held responsible for a failure of the team selection process

calebamiles commented:
/hold cancel

justaugustus commented:
@calebamiles -- addressed your feedback on the content. I note the escalation to SIG Release Chairs in the Timing section.

I'll leave the html for another day [and|or] person; it's making me feel... "eye-bleedy".

calebamiles commented on release-team/README.md:
```diff
@@ -188,6 +195,67 @@ For all unplanned or embargoed releases
+
+In the 1.11 release cycle, we selected the Release Team Lead by quorum during a public Release Team meeting and additional roles were staffed via [an issue in k/sig-release](https://github.com/kubernetes/sig-release/issues/167) and a set of PRs, starting with [this one](https://github.com/kubernetes/sig-release/pull/168).
+
+Each Release Team role is responsible for staffing their respective role, with this order of fall-through in mind:
```
nit: `s/Each Release Team role/Each Release Team member/`

calebamiles commented:
added a single comment but otherwise
/lgtm
